fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -78,6 +78,17 @@ def matching_build(watched_app_version: nil, watched_build_version: nil, app_id:
         watched_app_version_alternate = alternate_version(watched_app_version)
         versions = [watched_app_version, watched_app_version_alternate].compact
 
+        if versions.empty?
+          if select_latest
+            UI.message("Watched build version should not be present when there is no app version to watch") unless watched_build_version.nil?
+            UI.message("Searching for the latest build")
+            versions = [nil]
+          else
+            error_message = "FastlaneCore::BuildWatcher has no app version to watch"
```

Comment:
Maybe here is more convenient to raise the error instead of hardcoding the message

Reply:
Would it be fine to raise `FastlaneCore::Interface::FastlaneError` with
`UI.user_error!("There is no app version to watch")`?

Reply:
I mean if the error has done relation with build watcher you can always instantiate and raise that particular exception don't you?

Reply:
Sure, I'll add a `BuildWatcherError` and I'll also update raising error when more than 1 matching build is found in line 113.